python/pyspark/sql/observation.py

```diff
@@ -109,7 +111,9 @@ def _on(self, df: DataFrame, *exprs: Column) -> DataFrame:
         )
         return DataFrame(observed_df, df.sparkSession)
 
-    @property
+    # Note that decorated property only works with Python 3.9+ which Spark Connect requires.
+    @property  # type: ignore[misc]
+    @try_remote_observation
```

Comment:
what does this annotation mean?

Reply:
It calls the remote version of `Observation` API (see https://github.com/apache/spark/pull/39041/files/75bfce50110079a6918cb81169d1cd2197d56eba#diff-bdc591068d0c75d5011d05bf27cb6a49c3d31a583b60e0258059135e18881e98R352-R362).

Since `Observation` API is not implemented yet in Spark Connect, it throws an exception for now if `remote` is enabled.

Reply:
I see, so this is temporary?